### PR TITLE
feat: add node metadata.

### PR DESCRIPTION
**Pull request by lixd**
### What type of PR is this?

/kind feature

### What this PR does / why we need it:
add node metadata,config in  kubeclipper-agent.yaml report to kc-server by kc-agent.

### Which issue(s) this PR fixes:

Fixes #

### Special notes for reviewers:
```
1.kcctl deploy/join add --fip flags.
2.format kubeclipper-agent.yaml add metadata struct.
3.synchronization update deploy config cmd.
```

### Does this PR introduced a user-facing change?

```release-note
None
```

### Additional documentation, usage docs, etc.:

```docs

```

### CODE REVIEW

File: pkg/agent/config/config.go [hunk 1]
```diff
@@ -60,6 +60,11 @@ type Config struct {
 	ImageProxyOptions         *imageproxy.Options `json:"imageProxy,omitempty" yaml:"imageProxy,omitempty" mapstructure:"imageProxy"`
 }
 
+type MetaData struct {
+	Region string `json:"region,omitempty" yaml:"region,omitempty"`
+	FIP    string `json:"fip,omitempty" yaml:"fip,omitempty"`
```
Comment (zhuzhenfan) [inline pkg/agent/config/config.go:65]: 1. Suggest changing fip to floatIP or extranet IP.
2. add comment to explain

File: pkg/scheme/common/well_known_labels.go [hunk 2]
```diff
@@ -42,6 +42,7 @@ const (
 	LabelBackupPoint       = "kubeclipper.io/backupPoint"
 	LabelCronBackupDisable = "kubeclipper.io/cronBackupDisable"
 	LabelCronBackupEnable  = "kubeclipper.io/cronBackupEnable"
+	LabelMetadataFIP       = "metadata.kubeclipper.io/fip"
```
Comment (zhuzhenfan) [inline pkg/scheme/common/well_known_labels.go:45]: 1. Suggest changing fip to floatIP or extranet IP.

File: pkg/cli/config/template.go [hunk 3]
```diff
@@ -238,7 +238,11 @@ mq:
 
 const KcAgentConfigTmpl = `agentID: {{.AgentID}}
 ipDetect: {{.IPDetect}}
-region: {{.Region}}
+metadata:
+  region: {{.Region}}
+{{- if .FIP}}
```
Comment (zhuzhenfan) [inline pkg/cli/config/template.go:243]: 1. Suggest changing fip to floatIP or extranet IP.

File: cmd/kcctl/app/options/options.go [hunk 4]
```diff
@@ -161,12 +163,38 @@ type ImageProxy struct {
...
+}
+func (a Agents) Exists(ip string) bool {
+	_, ok := a[ip]
+	return ok
+}
+func (a Agents) Delete(ip string) {
+	delete(a, ip)
+}
+
+func (a Agents) Add(ip string, metadata Metadata) {
```
Comment (zhuzhenfan) [inline cmd/kcctl/app/options/options.go:183]: Variables using pointers

File: cmd/kcctl/app/options/options.go [hunk 5]
```diff
@@ -161,12 +163,38 @@ type ImageProxy struct {
...
+	for ip := range a {
+		list = append(list, ip)
+	}
+	return list
+}
+func (a Agents) Exists(ip string) bool {
+	_, ok := a[ip]
+	return ok
+}
+func (a Agents) Delete(ip string) {
```
Comment (zhuzhenfan) [inline cmd/kcctl/app/options/options.go:179]: Variables using pointers

File: cmd/kcctl/app/options/options.go [hunk 6]
```diff
@@ -181,24 +209,25 @@ type DeployConfig struct {
...
-func (a Agents) Add(region, ip string) {
+func (a AgentRegions) Add(region, ip string) {
 	if a.Exists(ip) {
 		return
 	}
 	a[region] = append(a[region], ip)
 }
 
-func (a Agents) Delete(ip string) {
+func (a AgentRegions) Delete(ip string) {
```
Comment (zhuzhenfan) [inline cmd/kcctl/app/options/options.go:230]: Variables using pointers

File: cmd/kcctl/app/options/options.go [hunk 7]
```diff
@@ -181,24 +209,25 @@ type DeployConfig struct {
...
+func (a AgentRegions) ListIP() []string {
 	set := sets.NewString()
 	for _, v := range a {
 		set.Insert(v...)
 	}
 	return set.List()
 }
 
-func (a Agents) Add(region, ip string) {
+func (a AgentRegions) Add(region, ip string) {
```
Comment (zhuzhenfan) [inline cmd/kcctl/app/options/options.go:223]: Variables using pointers

**Review by qinyer — APPROVED** [review]

**Review by x893675 — APPROVED** [review]

**Pull request merged**